app/models/user.rb

```diff
@@ -65,7 +65,7 @@ def report_registrable_from
   end
 
   def report_registrable_to
-    Time.current.last_month.since(5.days)
+    Time.current.since(5.days).last_month.to_date
```

Comment:
いえ、若干ややこしいところではあるのですが、「今日が何日なのか」って判定はTimezone考慮する必要があるので、 `Time.current` はそのまま残すようにしています。
「何日か」の判定を行った後は、入社日との比較を行えばいいだけなのでDate型にしました。

もし最初から無視するなら、
```
Date.today.since(5.days).last_month.to_date
```
みたいな書き方になっていたと思います。